homeworks/rostyslav.vyshemirskyi_Rooosss/PokemonProject/scripts/pokemons.js

```diff
@@ -309,14 +55,34 @@ function filterPokemons(pokemonsList) {
     return filter;
 }
 
+async function fetchData() {
+    try {
+        loading.style.display = 'block';
+        const options = {
+            method: 'GET',
+            mode: 'cors',
+            cache: 'no-cache',
+        };
+
+        const response = await fetch(baseUrl, options);
+        const json = await response.json();
+        pokemons = json.slice(0, 50);
+        renderCards(cardsList, pokemons);
+        loading.style.display = 'none';
+    } catch (error) {
+        console.error('GET error:', error.message);
+    }
+}
+
 function onSizeFilterChange() {
     renderCards(cardsList, filterPokemons(pokemons));
 }
 
-function init() {
+async function init() {
     bigPokemonCheckbox.addEventListener('change', onSizeFilterChange);
     smallPokemonCheckbox.addEventListener('change', onSizeFilterChange);
     renderCards(cardsList, pokemons);
+    await fetchData();
```

Comment:
Hey, @Rooosss, based on the name `fetchData` this function should only receive data.
Please return the data from this func and pass it to `renderCards`
Then you can skip rendering cards with 0 pokemons on the line 84

Reply:
Hey, @Rooosss, please don't resolve the comments by yourself - anyway, I should open them and check, have you fixed them or not.
So, once again, the name of the function states, that you receive data. And you like to render the cards inside it. It is unobvious.
Anybody (including you) should understand clearly, what some function does, without going to the function code. Just from the name of the function.
On the other hand, 1 function should do 1 action. So, please
> return the data from this func and pass it to `renderCards`